src/lewis/core/control_server.py

```diff
@@ -304,6 +304,7 @@ def start_server(self):
         if self._socket is None:
             context = zmq.Context()
             self._socket = context.socket(zmq.REP)
+            self._socket.setsockopt(zmq.RCVTIMEO, 100)
```

Comment:
If there's no timeout, it's not possible to stop the control server, because the thread hangs forever in recv. When there's a timeout, recv will fail after the timeout, raising EAGAIN, giving exceptions (such as `KeyboardInterrupt`) a chance to get noticed.